src/GraphQL.Tests/Utilities/FederatedSchemaBuilderTests.cs

```diff
@@ -156,5 +158,63 @@ public void input_types_and_types_without_key_directive_are_not_added_to_entitie
             
             Assert.Equal("User", name);
         }
+
+        [Fact]
+        public void resolve_reference_is_not_trying_to_await_for_each_field_individialy_and_plays_well_with_dataloader_issue_1565()
+        {
+            var definitions = @"
+                extend type Query {
+                    user(id: ID!): User
+                }
+
+                type User @key(fields: ""id"") {
+                    id: ID!
+                    username: String!
+                }
+            ";
+
+            var users = new List<User> {
+                new User { Id = "1", Username = "One" },
+                new User { Id = "2", Username = "Two" },
+            };
+
+            
+            var accessor = new DataLoaderContextAccessor
+            {
+                Context = new DataLoaderContext()
+            };
+            var listener = new DataLoaderDocumentListener(accessor);
+
+            Builder.Types.For("User").ResolveReferenceAsync(async ctx => {
+                var id = ctx.Arguments["id"].ToString();
+                // return Task.FromResult(users.FirstOrDefault(user => user.Id == id));
+                var loader = accessor.Context.GetOrAddBatchLoader<string, User>("GetAccountByIdAsync", ids => {
+                    var results = users.Where(user => ids.Contains(user.Id));
+                    return Task.FromResult((IDictionary<string, User>)results.ToDictionary(c => c.Id));
+                });
+                return await loader.LoadAsync(id);
+            });
+
+            var query = @"
+                {
+                    _entities(representations: [{__typename: ""User"", id: ""1"" }, {__typename: ""User"", id: ""2"" }]) {
+                        ... on User {
+                            id
+                            username
+                        }
+                    }
+                }";
+
+            var expected = @"{ ""_entities"": [{ ""__typename"": ""User"", ""id"" : ""1"", ""username"": ""One"" }, { ""__typename"": ""User"", ""id"" : ""2"", ""username"": ""Two"" }] }";
+
+            var executionResult = Executer.ExecuteAsync(_ =>
```

Comment:
I think this test should be able to be simplified by using `AssertQuery` vs. the `Executor` directly?

Reply:
Did tried this in first place, but because of `AssertQuery` lack of `Listeners` it was not possible, in a next commit I have added required `Listeners` to `ExecuteConfig` which solves both comments